optuna/storages/_rdb/storage.py

```diff
@@ -574,6 +585,7 @@ def _get_prepared_new_trial(
             trial.state = template_trial.state
 
         trial.number = trial.count_past_trials(session)
+        self.record_heartbeat(trial.trial_id)
```

Comment:
This line is necessary because, without it, the table `trial_heartbeats` will not be created at the beginning of `Study.optimize`. However, as you said, we can move the process of recording heartbeat to `_optimize.py`, so I will do that.